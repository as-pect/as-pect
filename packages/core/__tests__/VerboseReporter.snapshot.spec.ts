import { createSnapshotModule } from "./setup/createSnapshotModule";
import { VerboseReporterWrapper } from "./setup/VerboseReporterWrapper";

const reporter = new VerboseReporterWrapper();

beforeAll(
  () =>
    new Promise((resolve, reject) => {
      createSnapshotModule(
        {},
        (err, _result) => {
          if (err) {
            console.log(err);
            reject(err);
          } else {
            resolve();
          }
        },
        reporter,
      );
    }),
);

test("snapshots", () => {
  for (const [name, values] of reporter.snapshots.entries()) {
    for (const value of values) {
      expect(value).toMatchSnapshot(name);
    }
  }
});
