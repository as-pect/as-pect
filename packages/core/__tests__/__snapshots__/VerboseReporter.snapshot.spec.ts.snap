// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots: onFinish assembly/jest-reporter-snapshot.ts 1`] = `
"[Snapshot]: a few snapshots!~first vec3[0]
  Vec3 {
    x: 1.0,
    y: 2.0,
-   z: 4.0,
-   a: 10.0
+   z: 3.0
  }

[Snapshot]: a few snapshots!~some integer[0]
+ 504

[Snapshot]: this!~is!~removed
- false

    [File]: assembly/jest-reporter-snapshot.ts RTrace: +8
  [Groups]: 3 pass, 3 total
  [Result]: ✖ FAIL
[Snapshot]: 5 total, 1 added, 1 removed, 1 different
 [Summary]: 2 pass,  0 fail, 2 total
    [Time]: 0ms

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

"
`;

exports[`snapshots: onGroupEnd  1`] = `""`;

exports[`snapshots: onGroupEnd logs 1`] = `
"    [Todo]: one
    [Todo]: two
    [Todo]: three

"
`;

exports[`snapshots: onGroupEnd snapshots 1`] = `
"
"
`;

exports[`snapshots: onGroupStart logs 1`] = `
"[Describe]: logs

     [Log]: \\"Hello world!\\"
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/string/String> (wasm-function[49]:0x170a)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x765c)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 42
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i32> (wasm-function[53]:0x17bd)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x7660)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: <Vec3>null
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Vec3 | null> (wasm-function[66]:0x24ca)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x7664)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Vec3 {
                x: 1.0,
                y: 2.0,
                z: 3.0
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Vec3> (wasm-function[70]:0x260e)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x7687)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<i32> [
                1,
                2,
                3
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<i32>> (wasm-function[76]:0x27a2)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x7698)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 9999999999999 as i64
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i64> (wasm-function[80]:0x2855)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76a2)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 1234567 as u32
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<u32> (wasm-function[84]:0x2904)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76a9)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 999999999999 as u64
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<u64> (wasm-function[88]:0x29b3)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76b2)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: -1 as i8
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i8> (wasm-function[92]:0x2a62)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76b6)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: -1 as i16
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i16> (wasm-function[96]:0x2b11)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76ba)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: true
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<bool> (wasm-function[100]:0x2bc0)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76be)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: false
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<bool> (wasm-function[100]:0x2bc0)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76c2)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Uint8Array [ 0, 1, 2 ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/typedarray/Uint8Array> (wasm-function[107]:0x2d6c)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76c6)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Int8Array [ -1, -2, -3 ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/typedarray/Int8Array> (wasm-function[112]:0x2ebf)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76ca)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: [Function 1: ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<(i: i32) => i32> (wasm-function[115]:0x2f79)
            at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x76ce)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: \\"trace: The thing! 1, 2, 3, 4, 5\\"
   [Stack]: at start:assembly/jest-log~anonymous|0 (wasm-function[304]:0x7702)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
"
`;

exports[`snapshots: onGroupStart snapshots 1`] = `
"[Describe]: snapshots

"
`;

exports[`snapshots: onTestFinish logs logs 1`] = `
" [Success]: ✔ logs RTrace: +4
     [Log]: \\"Hello world!\\"
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/string/String> (wasm-function[49]:0x170a)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x727b)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: \\"Calculated 42\\"
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/string/String> (wasm-function[49]:0x170a)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x728a)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 42
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i32> (wasm-function[53]:0x17bd)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x728e)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: <Vec3>null
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Vec3 | null> (wasm-function[66]:0x24ca)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7292)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Vec3 {
                x: 1.0,
                y: 2.0,
                z: 3.0
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Vec3> (wasm-function[70]:0x260e)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72b5)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<i32> [
                1,
                2,
                3
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<i32>> (wasm-function[76]:0x27a2)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72c6)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 9999999999999 as i64
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i64> (wasm-function[80]:0x2855)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72d0)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 1234567 as u32
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<u32> (wasm-function[84]:0x2904)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72d7)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: 999999999999 as u64
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<u64> (wasm-function[88]:0x29b3)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72e0)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: -1 as i8
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i8> (wasm-function[92]:0x2a62)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72e4)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: -1 as i16
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<i16> (wasm-function[96]:0x2b11)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72e8)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: true
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<bool> (wasm-function[100]:0x2bc0)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72ec)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: false
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<bool> (wasm-function[100]:0x2bc0)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72f0)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Uint8Array [ 0, 1, 2 ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/typedarray/Uint8Array> (wasm-function[107]:0x2d6c)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72f4)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Int8Array [ -1, -2, -3 ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/typedarray/Int8Array> (wasm-function[112]:0x2ebf)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72f8)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: [Function 1: ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<(i: i32) => i32> (wasm-function[115]:0x2f79)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x72fc)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: \\"trace: The thing! 1, 2, 3, 4, 5\\"
   [Stack]: at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7330)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: UnmanagedClass {
                a: 1.0,
                b: 2.0,
                c: 3.0
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/UnmanagedClass> (wasm-function[127]:0x33f3)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x733a)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<i32> [
                0,
                1,
                2,
                3,
                4,
                5,
                6,
                7,
                8,
                9,
                10,
                11,
                12,
                13,
                14,
                15,
                16,
                17,
                18,
                19
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<i32>> (wasm-function[76]:0x27a2)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7376)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: LotsOfFields {
                a: 1.0,
                b: 1.0,
                c: 1.0,
                d: 1.0,
                e: 1.0,
                f: 1.0,
                g: 1.0,
                h: 1.0,
                i: 1.0,
                j: 1.0,
                k: 1.0,
                l: 1.0,
                m: 1.0,
                n: 1.0,
                o: 1.0,
                p: 1.0,
                q: 1.0,
                r: 1.0,
                s: 1.0,
                t: 1.0,
                u: 1.0,
                v: 1.0,
                w: 1.0,
                x: 1.0,
                y: 1.0,
                z: 1.0
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/LotsOfFields> (wasm-function[140]:0x3d93)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x737f)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Float64Array [ 0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0... +10 items ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/typedarray/Float64Array> (wasm-function[147]:0x3f44)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x73b9)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<i32> [
                1,
                2,
                3
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<i32>> (wasm-function[76]:0x27a2)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x73cb)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Circular {
                a: 1.0 as f32,
                b: [Circular Reference],
                c: 3.0 as f32,
                d: <Array<assembly/jest-log/Circular>>null
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Circular> (wasm-function[158]:0x4382)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x73fb)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<assembly/jest-log/Circular> [
                {
                    a: 1.0 as f32,
                    b: <Circular>null,
                    c: 3.0 as f32,
                    d: [Circular Reference]
                }
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<assembly/jest-log/Circular>> (wasm-function[163]:0x4540)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x743d)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedA {
                a: NestedB {
                    b: NestedC {
                        c: [NestedD]
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedA> (wasm-function[183]:0x4c46)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7447)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Map<assembly/jest-log/Vec3,i32> {
                [Vec3]: 44
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/map/Map<assembly/jest-log/Vec3,i32>> (wasm-function[201]:0x5384)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x74d7)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: [Function 2: ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<() => ~lib/string/String> (wasm-function[205]:0x5447)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x74dc)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedForInlineFunctionA {
                a: NestedForInlineFunctionB {
                    b: NestedForInlineFunctionC {
                        c: [Function]
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedForInlineFunctionA> (wasm-function[218]:0x585d)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x74e6)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedForInlineStringA {
                a: NestedForInlineStringB {
                    b: NestedForInlineStringC {
                        c: \\"The Inlined String\\"
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedForInlineStringA> (wasm-function[229]:0x5c0f)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x74f0)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedForInlineU8A {
                a: NestedForInlineU8B {
                    b: NestedForInlineU8C {
                        c: 42 as u8
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedForInlineU8A> (wasm-function[240]:0x5fc0)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x74fa)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedForInlineBoolA {
                a: NestedForInlineBoolB {
                    b: NestedForInlineBoolC {
                        c: true
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedForInlineBoolA> (wasm-function[251]:0x6371)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7504)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedForInlineBoolA {
                a: NestedForInlineBoolB {
                    b: NestedForInlineBoolC {
                        c: false
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedForInlineBoolA> (wasm-function[251]:0x6371)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x751d)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedTypeImpliedArrayA {
                a: NestedTypeImpliedArrayB {
                    b: NestedTypeImpliedArrayC {
                        c: Array<~lib/array/Array<u8>> [ [ 0 ] ]
                    }
                }
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedTypeImpliedArrayA> (wasm-function[268]:0x6944)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7527)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: NestedTypeImpliedArrayC {
                c: Array<~lib/array/Array<u8>> [
                    [
                        0
                    ]
                ]
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/NestedTypeImpliedArrayC> (wasm-function[270]:0x69ab)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x7531)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<assembly/jest-log/Vec3> [
                {
                    x: 1.0,
                    y: 2.0,
                    z: 3.0
                },
                {
                    x: 1.0,
                    y: 2.0,
                    z: 3.0
                },
                {
                    x: 1.0,
                    y: 2.0,
                    z: 3.0
                },
                {
                    x: 1.0,
                    y: 2.0,
                    z: 3.0
                }
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<assembly/jest-log/Vec3>> (wasm-function[274]:0x6b3d)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x758c)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Array<assembly/jest-log/Vec3 | null> [
                null
            ]
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<~lib/array/Array<assembly/jest-log/Vec3 | null>> (wasm-function[281]:0x6d15)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x75a2)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: MeaningOfLife {
                its: 42
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Overridden> (wasm-function[289]:0x6f2f)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x75ac)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
     [Log]: Child {
                a: 0,
                c: 2,
                b: 4,
                d: 6
            }
   [Stack]: at ../assembly/assembly/internal/Reflect/Reflect.attachStackTrace (wasm-function[48]:0x16dc)
            at ../assembly/assembly/internal/log/log<assembly/jest-log/Child> (wasm-function[299]:0x7268)
            at start:assembly/jest-log~anonymous|0~anonymous|0 (wasm-function[300]:0x75b6)
            at ../assembly/assembly/internal/call/__call (wasm-function[327]:0x7aca)
"
`;

exports[`snapshots: onTestFinish snapshots a few snapshots 1`] = `
" [Success]: ✔ a few snapshots
"
`;

exports[`snapshots: onTestStart logs logs 1`] = `""`;

exports[`snapshots: onTestStart snapshots a few snapshots 1`] = `""`;
